codegen/src/main/java/software/amazon/awssdk/codegen/poet/client/DelegatingAsyncClientClass.java

```diff
@@ -42,42 +49,95 @@ public class DelegatingAsyncClientClass extends AsyncClientInterface {
     public DelegatingAsyncClientClass(IntermediateModel model) {
         super(model);
         this.model = model;
-        this.className = ClassName.get(model.getMetadata().getFullInternalPackageName(),
+        this.className = ClassName.get(model.getMetadata().getFullClientPackageName(),
                                        "Delegating" + model.getMetadata().getAsyncInterface());
         this.poetExtensions = new PoetExtension(model);
     }
 
     @Override
-    public TypeSpec poetSpec() {
+    protected TypeSpec.Builder createTypeSpec() {
+        return PoetUtils.createClassBuilder(className);
+    }
+
+    @Override
+    protected void addInterfaceClass(TypeSpec.Builder type) {
         ClassName interfaceClass = poetExtensions.getClientClass(model.getMetadata().getAsyncInterface());
-        TypeSpec.Builder result = PoetUtils.createClassBuilder(className);
 
-        result.addSuperinterface(interfaceClass)
-              .addAnnotation(SdkInternalApi.class)
-              .addModifiers(Modifier.ABSTRACT, Modifier.PUBLIC)
-              .addField(FieldSpec.builder(interfaceClass, "delegate")
-                                 .addModifiers(Modifier.PROTECTED, Modifier.FINAL)
-                                 .build())
-              .addMethods(operations())
-              .addMethod(closeMethod());
+        MethodSpec delegate = MethodSpec.methodBuilder("delegate")
+                                        .addModifiers(PUBLIC)
+                                        .addStatement("return this.delegate")
+                                        .returns(SdkClient.class)
+                                        .build();
+
+        type.addSuperinterface(interfaceClass)
+            .addMethod(constructor(interfaceClass))
+            .addField(FieldSpec.builder(interfaceClass, "delegate")
+                               .addModifiers(PRIVATE, FINAL)
+                               .build())
```

Comment:
nit: should this be in addFields?

Reply:
moved to addFields, same for DelegatingSyncClientClass